src/main/resources/static/js/svgradiobutton.js

```diff
@@ -0,0 +1,25 @@
+class RadioButtonManager {
+    constructor(radioGroupName, svgClass) {
+        this.radioButtons = document.querySelectorAll(`input[name="${radioGroupName}"]`);
+        this.svgElements = document.querySelectorAll(`.${svgClass}`);
+        if (this.radioButtons.length > 0 && this.svgElements.length > 0) {
+            this.init();
+        }
+    }
+    init() {
+        this.radioButtons.forEach(radio => {
+            radio.addEventListener('change', () => this.updateClasses());
+        });
+        this.updateClasses();
+    }
+    updateClasses() {
+        this.radioButtons.forEach((radio, index) => {
+            if (radio.checked) {
+                this.svgElements[index].classList.remove('fill-label');
+            } else {
+                this.svgElements[index].classList.add('fill-label');
+            }
+        });
```

Comment:
やるならこんな感じかな
動確してないからいい感じに書き換えてね
```suggestion
class RadioButton {
    #element;
    #svgElement

    constructor(element, svgElement) {
        this.#element = element;
        this.#svgElement = svgElement;
        this.#element.addEventListener('change', () => this.onChanged());
        init();
    }
    
    onChanged() {
        if (radio.checked) {
            this.#svgElement.classList.remove('fill-label');
            return;
        }
        this.#svgElement.classList.add('fill-label');
    }
}


class RadioButtonList {
    #values = new Array(0);
    
    constructor(radioGroupName, svgClass) {
        const radioButtonElements = document.querySelectorAll(`input[name="${radioGroupName}"]`);
        const svgElements = document.querySelectorAll(`.${svgClass}`);
        
        if (this.#values .length <= 0 || svgElements .length <= 0) return;
        
        this.radioButtons.forEach((radioButton, index) => {
            const radioButton = new RadioButton(radioButton, svgElements[index]);
            this.#values.push(radioButton);
        });
    }
```

Reply:
[9d45df4](https://github.com/saitokazumasa/tabisketch/pull/148/commits/9d45df488e3e8a190933fc1105c7059edefa7b17)
[4666bf8](https://github.com/saitokazumasa/tabisketch/pull/148/commits/4666bf82a66453fa1653e9218b678438bfce66f8)
修正しました。

Reply:
RadioButtonListがRadioButtonに依存している方向性で、
RadioButtonがRadioButtonに依存する書き方は良くないかな。
依存の方向を統一して欲しい。一般的には大きい概念から小さい概念に依存するとすっきりするかも。
※めっちゃ要約して書くと、RadioButtonでRadioButtonListを操作しないでってこと
```
class RadioButton {
    #parentList;
    constructor() {
        this.#element.addEventListener('change', () => this.#parentList.checkAll()); // NG
    }
}
```

多分RadioButtonのどれかが操作されたときに動かす関数っていう意味で`onChecked()`にしたんだと思うんだけど、
これってRadioButtonの`onChanged()`が動いたときに一緒に動かしているだけで、実際は変更に伴って動いてる関数ではないよね。
そういうときは`on○○`より、こういう操作をしている、っていう関数名にしよう。
```
class RadioButtonList {
    onChecked() {
        this.#values.forEach((radioButton) => {
            radioButton.onChanged();
        });
}
```

Reply:
[9a5bed8](https://github.com/saitokazumasa/tabisketch/pull/148/commits/9a5bed863f56e37200bd1ab90400d90a8e53cfe5)
修正しました。

Reply:
う～ん、変わってないかなぁ
依存の意味をもう一度調べてみて欲しいんだけど、そもそもRadioButtonがRadioButtonListをコンストラクタの引数として受け取って保持している時点で逆方向に依存しちゃってるんだよね。

Reply:
[88b40b7](https://github.com/saitokazumasa/tabisketch/pull/148/commits/88b40b7568a82c17e3edcd284e511449cf6ba885)
かなり変えちゃいましたが修正しました。

Reply:
サンクス！
別コメントしたため、このコメントは解決。